### ISSUE TITLE
Turn the metadate.md file into a table and create a generator to auto generate the MD file

### ISSUE THREAD

**Issue by hswerdfe** [Big-Life-Lab/covid-19-wastewater#95]
Proposed columns for the table

 * **tableName**: 
 * **tableDesc**: description of the table, but only populated for the first row for each table
 * **columnName**:  
 * **columnDesc**: description of the column but only populated for the first column of each table
 * **ColumnType**: type information this columns should follow
 * **columnValues** : for categorical data it would have a  value that is allowed
 * **columnValuesDescriptions** : for categorical data it a descriptions of the values
 * **columnValidation**: would have a small amount of executable R or Python code to execute on the data

**Comment by DougManuel** [comment]
Hmm, suggest splitting into three tables.
- tables: tableName, tableDesc
- variables: all the `column`s described above. I think we also need a `key` header.
- variable_categories: category values for variables: `name` and `description`. On the balance, I think we should have a both and name and a value, but I don't feel strongly about that. People will likely create a value if we don't.

**Comment by hswerdfe** [comment]
@DougManuel sounds good I'll add a file to DEV branch, so we can review soon.

**Comment by DougManuel** [comment]
Many similar metadata libraries will also have a unique ID for each variable. V1, V2, etc. I don't think that is absolutely necessary, but raising the issue.

**Comment by hswerdfe** [comment]
I am working it out now. One issue is some of the formating in the description like this.

Links with the `AssayMethod` used to perform the analysis. Use `instrument.ID` for measures that are not viral measures.

I am not sure what to do with this. We can either 
 * remove the formatting 
 * include md formatting in the descriptions (this could cause problems with escape characters 
 * we could also auto detect column names when generating the metadata file and add the formatting.

**Comment by DougManuel** [comment]
We started to take out some of this info as we moved towards V1.0. 
- We probably want to split some of this info up into different headers. We probably want a variable label and also description, notes or other instructions. A use case of a variable label is in a descriptive table and normally wouldn't include instructions or context.
- I don't think we should have special characters for most if not all attributes. That would be unusual for metadata. Sometimes special characters appear in  longer descriptive fields, such as an table abstract.
